test/inductor/test_foreach.py

```diff
@@ -930,6 +984,77 @@ def fn(a0, a1, a2, b0, b1, b2):
 
         self.assertEqual(torch._inductor.metrics.generated_kernel_count, 1)
 
+    @requires_cuda
+    @foreach_map_bin_ops
+    def test_foreach_map_backward_binary(self, op):
+        from torch._dynamo.polyfills import foreach_map_fn
+
+        def fn(xs, ys):
+            outs = op(xs, ys)
+            return outs[0].sum() + outs[1].sum() + outs[2].sum()
+
+        def ref_fn(xs, ys):
+            outs = foreach_map_fn(torch.add, xs, ys)
+            return outs[0].sum() + outs[1].sum() + outs[2].sum()
+
+        ref_inps = (
+            [
+                torch.rand(10, 20, device="cuda:0", requires_grad=True),
+                torch.rand(10, 30, device="cuda:0", requires_grad=True),
+                torch.rand(30, 30, device="cuda:0", requires_grad=True),
+            ],
+            [
+                torch.rand(10, 20, device="cuda:0", requires_grad=True),
+                torch.rand(10, 30, device="cuda:0", requires_grad=True),
+                torch.rand(30, 30, device="cuda:0", requires_grad=True),
+            ],
+        )
+        inps = (
+            [x.clone().detach().requires_grad_(True) for x in ref_inps[0]],
+            [y.clone().detach().requires_grad_(True) for y in ref_inps[1]],
+        )
+
+        out_ref = ref_fn(*ref_inps)
+        out_ref.backward()
+
+        fn_opt = torch.compile(fn, fullgraph=True)
+        out = fn_opt(*inps)
+        out.backward()
```

Comment:
Can you check for the number of kernels generated in backward to test we are foreaching backward ? `run_and_get_code` will give you code for both fwd/backward.